### DISCUSSION TITLE
Chat messages encryption with pgsodium makes inserts impossible : 'query returned no rows'

### DISCUSSION THREAD

**Discussion by AdelieC** [supabase/supabase#15150] [Questions]
Hello ! I have a table `messages` with an encrypted column of type text `content`. I set the encryption on it this way :
`
SECURITY LABEL FOR pgsodium
	ON COLUMN messages.content is 'ENCRYPT WITH KEY COLUMN room_id';
`
As you can see, i want my encryption key to be the id of the room the message belongs to. room_id is a foreign key of type uuid.

I also granted execute to authenticated on pgsodium functions, like this :
`
GRANT EXECUTE ON FUNCTION pgsodium.crypto_aead_det_encrypt (bytea, bytea, uuid, bytea) TO authenticated;
GRANT EXECUTE ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, uuid, bytea) TO authenticated;
`
And i have policies on both `messages` and `rooms` to allow select and insert to authenticated.

**PROBLEM** : Now when i try to insert into messages, i get this error : 'Failed to run sql query: query returned no rows'
**OTHER PROBLEM** : when i try to disable the trigger `messages_encrypt_secret_trigger_content` from the dashboard, i get this error: 'Failed to create trigger: failed to update pg.triggers with the given ID: Cannot find a trigger with ID 28950'

Any idea why ? What did i do wrong ?

**Comment by AdelieC** [comment]
Update : i tried several other things to understand what the problem could be :

1. i tried with this security label :
```
SECURITY LABEL FOR pgsodium
	ON COLUMN public.messages.content is 'ENCRYPT WITH KEY ID b115763d-e5f1-4299-82aa-c934b6172790 ASSOCIATED (room_id)';
```
2. i tried the query in the doc :
```
select * from pgsodium.crypto_aead_det_encrypt (
  'this is the message',  -- a message to encrypt
  'this is associated data',  -- some authenticated associated data
  'eaa20d8c-c77c-4985-9f73-2a5f5d1f1e6d'::uuid -- key ID
);
```
3. I tried this query like this :
```
select * from pgsodium.crypto_aead_det_encrypt(
    pg_catalog.convert_to('blah', 'utf8'),
    pg_catalog.convert_to(('')::text, 'utf8'),
    'b115763d-e5f1-4299-82aa-c934b6172790'::uuid,
    NULL
);
```

All of those present the same issue : `query returned no rows`

I checked privileges for authenticated and dashboard_user, and my grant commands worked since i have them in pgsodium_keyholder results. And which is more, they have execute on pgsodium encrytion/decryption functions, i made sure of this. I'm at a loss.

**Reply by jonkurtis** [reply]
Where did you get the uuid value used here:
```
SECURITY LABEL FOR pgsodium
	ON COLUMN public.messages.content is 'ENCRYPT WITH KEY ID b115763d-e5f1-4299-82aa-c934b6172790 ASSOCIATED (room_id)';
```
I suspect that this uuid is not referencing a pgsodium key. You need to go create one by running:
`select * from pgsodium.create_key();` and then using the returning uuid in the security label.

You can switch your table schema to `pgsodium` instead of `public` in the GUI and check the `pgsodium.key` table and verify that the uuid you are using in the security label is in that table. If it isn't, then it won't work.

**Comment by sumatoken** [comment]
I am facing the same issue. I've enabled TCE to encrypt a certain column but now I. can't insert into my table, even from the dashboard GUI.  I get the same error as you

**Comment by devbydaniel** [comment]
Same here. I enabled column encryption for a certain column and get the same error when inserting data into the table with service_user role or via GUI, but only on localhost. On my remote project with the same database schema, inserting records via the GUI works. Any ideas?

**Comment by anthonyverducci** [comment]
Hi everyone, I've submitted a bug report for this issue: #18192
